plutarch-testlib/Plutarch/Test/Laws.hs

```diff
@@ -311,6 +313,25 @@ checkPLiftableLaws =
       plift (pconstant @a x) `prettyEquals` x
   ]
 
+class AsTagTest a where
+  tagNum :: a -> Integer
```

Comment:
I see what you were going for there, but I don't think it's the best idea. A type class like this, only for testing purposes, won't work terribly well, because any type someone would want to test would need to be an instance.

It'd be better to do this using `SOP.Generic`, since we need an instance of `SOP.Generic` for Plutarch types anyway.